roles/docker/tasks/main.yml

```diff
@@ -9,105 +9,38 @@
     - docker
     - bootstrap
 
-- name: install docker packages
-  sudo: yes
-  yum:
-    name: "{{ item }}"
-    state: present
-  with_items:
-    - docker
-    - docker-selinux
-  tags:
-    - docker
-    - bootstrap
-
-- name: check if docker storage already set up with devicemapper
-  sudo: yes
-  shell: "test -f /etc/sysconfig/docker-storage-setup && . /etc/sysconfig/docker-storage-setup && echo ${STORAGE_DRIVER:-NOT_SET}"
-  register: docker_storage_already_set_up
-  tags:
-    - disk
-    - docker
-
-- name: set docker_configure_lvm
-  set_fact:
-    docker_configure_lvm: '{{ docker_lvm_backed|bool and docker_storage_driver in ["btrfs", "overlay"] and docker_storage_already_set_up.stdout in ["NOT_SET", "btrfs", "overlay"] }}'
-  tags:
-    - disk
-    - docker
-
-- name: create backing storage volume for docker storage
-  sudo: yes
-  lvol:
-    vg: "{{ volume_group_name }}"
-    lv: "{{ docker_volume_name }}"
-    size: "{{ docker_volume_size }}"
-  when: docker_configure_lvm|bool
-  tags:
-    - disk
-    - docker
-
-- name: create backing storage filesystem for docker storage
+- name: enable Docker, Inc. yum repo
   sudo: yes
-  filesystem:
-    dev: "{{ docker_volume_device }}"
-    fstype: "{{ docker_volume_fs_type }}"
-    opts: "{{ docker_volume_fs_opts }}" 
-  when: docker_configure_lvm|bool
-  tags:
-    - disk
-    - docker
-
-- name: create mount point for docker storage
-  sudo: yes
-  file:
-    state: directory
-    name: "{{ docker_volume_mount }}"
-    mode: 0755
-  when: docker_configure_lvm|bool
+  copy:
+    src: docker.repo
+    dest: /etc/yum.repos.d/docker.repo
   tags:
-    - disk
     - docker
+  when: docker_package_source is defined and docker_package_source == "docker" 
 
-- name: mount docker storage
+- name: remove old docker package
   sudo: yes
-  mount:
-    state: mounted
-    name: "{{ docker_volume_mount }}"
-    fstype: "{{ docker_volume_fs_type }}"
-    src: "{{ docker_volume_device }}"
-    dump: 1
-    passno: 2
-  when: docker_configure_lvm|bool
+  yum:
+    name: "{{ item }}"
+    state: absent
```

Comment:
Manual need restart on upgrade.
Currently docker have started=yes in ansible, and it need to be switchable off to packer